samples/gibbs_ensemble_socket.py

```diff
@@ -0,0 +1,437 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Simulate a Gibbs-ensemble of a Lennard-Jones fluid at a fixed temperature.
+This script does the Monte-Carlo part of the Gibbs-ensemble, however for the
+energy calculation of the systems, two instances of the `gibbs_ensemble_client.py`
+script are executed, which each run a different instance of ESPResSo.
+
+The Gibbs-ensemble implemented in these scipts closely refers to chapter 8 of
+Daan Fenkel and Berend Smit 'Understanding Molecular Simulation, Second edition'.
+All equation references noted in this script can be found there.
+
+Due to the cutoff and shifting of the LJ-potential the simulated points in the
+phase diagram differ from the long range uncut LJ-potential. The phase diagram
+of the used potential can be found in 'B. Smit, J. Chem. Phys. 96 (11), 1992,
+Phase diagrams of Lennard-Jones fluids'.
+"""
+
+import socket
+import numpy as np
+import pickle
+import subprocess
+import struct
+import random
+import matplotlib.pyplot as plt
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--seed', type=int, nargs=1)
+parser.add_argument('-S', '--steps', type=int, nargs=1)
+parser.add_argument('-E', '--espresso-executable', nargs=1)
+parser.add_argument('-C', '--client-script', nargs=1)
+parser.add_argument('-n', '--number-of-particles', type=int, nargs=1)
+parser.add_argument('-l', '--box-length', type=float, nargs=1)
+parser.add_argument('-T', '--temperature', type=float, nargs=1)
+
+args = parser.parse_args()
+
+# system parameters
+steps = 50000
```

Comment:
please use all-caps style for ```CONSTANTS```

Reply:
should constants which can be defined via argparse arguments still count as constants?

Reply:
no they are variables